acestep/mlx_dit/generate.py

```
@@ -146,8 +151,30 @@ def mlx_generate_diffusion(
+        def _raw_step(xt, t, tr, enc, ctx):
+            vt, _ = mlx_decoder(
+                hidden_states=xt, timestep=t, timestep_r=tr,
+                encoder_hidden_states=enc, context_latents=ctx,
+                cache=None, use_cache=False,
+            )
+            return vt
+
+        try:
+            _compiled_step = mx.compile(_raw_step)
+            logger.info("[MLX-DiT] Diffusion step compiled with mx.compile().")
+        except Exception as exc:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)